lib/slack/web/pagination/cursor.rb

```diff
@@ -7,23 +7,31 @@ class Cursor
 
           attr_reader :client
           attr_reader :verb
+          attr_reader :pause
           attr_reader :params
 
           def initialize(client, verb, params = {})
             @client = client
             @verb = verb
+            @pause = params.delete(:pause)
             @params = params
           end
 
           def each
             next_cursor = nil
             loop do
               query = { limit: client.default_page_size }.merge(params).merge(cursor: next_cursor)
-              response = client.send(verb, query)
+              begin
+                response = client.send(verb, query)
+              rescue Slack::Web::Api::Errors::TooManyRequestsError => e
+                sleep(e.retry_after.seconds)
+                next
```

Comment:
I also think we should have a max retry count with some default set fairly high. If slack blacklists you somehow you'll have code that never ends and it will be impossible to debug.

I also think there should be a log line here in DEBUG mode.

Reply:
Do you think the count should reset after a successful request?

Reply:
I think so.